multiqc/plots/plotly/plot.py

```diff
@@ -1079,26 +1079,53 @@ def __control_panel(self, flat: bool, module_anchor: Anchor, section_anchor: Anc
         return html
 
 
-def _export_plot(fig, file_ext, plot_path, write_kwargs) -> Optional[str]:
-    if is_running_under_rosetta():
-        return None
+def _export_plot(fig, file_ext, plot_path, write_kwargs):
+    """Export a plotly figure to a file."""
+    import logging
+    import signal
+    import threading
+    from functools import partial
 
-    try:
-        if file_ext == "svg":
-            # Cannot add logo to SVGs
-            fig.write_image(plot_path, **write_kwargs)
-        else:
-            img_buffer = io.BytesIO()
-            fig.write_image(img_buffer, **write_kwargs)
-            img_buffer = add_logo(img_buffer, format=file_ext)
-            with open(plot_path, "wb") as f:
-                f.write(img_buffer.getvalue())
-            img_buffer.close()
-    except Exception as e:
-        logger.error(f"Unable to export plot to {file_ext.upper()} image: {e}")
-        return None
-    else:
-        return plot_path
+    # Default timeout of 30 seconds for image export
+    timeout = config.export_plots_timeout if hasattr(config, "export_plots_timeout") else 30
+
+    def timeout_handler(*args):
+        raise TimeoutError(f"Plot export timed out after {timeout} seconds")
+
+    class ExportThread(threading.Thread):
+        def __init__(self, fig, plot_path, write_kwargs):
+            threading.Thread.__init__(self)
+            self.fig = fig
+            self.plot_path = plot_path
+            self.write_kwargs = write_kwargs
+            self.exception = None
+
+        def run(self):
+            try:
+                self.fig.write_image(self.plot_path, **self.write_kwargs)
+            except Exception as e:
+                self.exception = e
+
+    # Start the export in a separate thread
+    export_thread = ExportThread(fig, plot_path, write_kwargs)
+    export_thread.start()
+    export_thread.join(timeout)
+
+    if export_thread.is_alive():
+        # If thread is still running after timeout, log warning and continue
+        logging.warning(
+            f"Plot export timed out after {timeout}s: {plot_path}. "
+            "This is likely due to a known issue in Kaleido. "
+            "The plot will be skipped but the report will continue to generate."
+        )
+        # We don't kill the thread as it might be unsafe, but we continue execution
```

Comment:
I think you may have to kill it eventually, this might be the reason why the program gets stuck on "multiqc completed"